datafusion/core/src/physical_plan/file_format/csv.rs

```diff
@@ -252,7 +252,13 @@ pub async fn plan_to_csv(
                 });
                 tasks.push(handle);
             }
-            futures::future::join_all(tasks).await;
+            let results = futures::future::join_all(tasks).await;
+            for result in results {
+                let result =
+                    result.map_err(|e| DataFusionError::Execution(format!("{}", e)))?;
+                let _ =
+                    result.map_err(|e| DataFusionError::Execution(format!("{}", e)))?;
+            }
```

Comment:
open to suggestions on a better implementation here

Reply:
I suppose some tests would be good too ... I will add some to this PR later today

Reply:
I think we can use `try_for_each` from https://docs.rs/futures-util/0.3.24/futures_util/stream/trait.TryStreamExt.html#method.try_for_each

Reply:
That worked well. Thank you.